### PR TITLE
Spectra plot improve

### PR DESCRIPTION
**Pull request by grburgess**
This moves the SpectrumLike plotting to a member function and allows for over plotting of other models (such as was seen in Vianello et al 2017 ;) ).

The old global function was kept, but is simple a convience wrapper that allows for selecting color sets, etc. 

A few other features were added to improve plotting.

### CODE REVIEW

**Comment by giacomov** [comment]
It looks like you are still working on this. Please tell me when you are done and I'll merge it.

Does it change anything regarding the documentation? Please have a look and make sure that the documentation still works.

**Comment by grburgess** [comment]
Ah, I just added some docs. 

Did you add 3 docs for this?

**Comment by grburgess** [comment]
Documentation was added. In the process, I added .from_function class methods to the SpectrumLike classes following the XYLike approach. This should be useful for testing the background modeling in the near future.

**Comment by grburgess** [comment]
@giacomov I am done with this for now. I'll continue to work on the documentation but add that directly .

File: threeML/plugins/DispersionSpectrumLike.py [hunk 1]
```diff
@@ -123,3 +127,121 @@ def write_pha(self, filename, overwrite=False, force_rsp_write=False):
 
         ogiplike = OGIPLike.from_general_dispersion_spectrum(self)
         ogiplike.write_pha(file_name=filename, overwrite=overwrite, force_rsp_write=force_rsp_write)
+
```
Comment (giacomov) [inline threeML/plugins/DispersionSpectrumLike.py]: Nice!

File: threeML/plugins/SpectrumLike.py [hunk 2]
```diff
@@ -234,6 +236,122 @@ def __init__(self, name, observation, background, verbose=True):
         # Apply the mask
         self._apply_mask_to_original_vectors()
 
+    @classmethod
```
Comment (giacomov) [inline threeML/plugins/SpectrumLike.py]: It looks like this shares a lot of code with the other .from_function method you wrote in this pull request.

Would it be possible to put the common code between the two somewhere else (maybe the parent class)?

**Comment by grburgess** [comment]
@giacomov I removed the duplicated code and added some tests to check fitting of the SpectrumLike classes now that we have generators.

**Comment by giacomov** [comment]
Thanks! Merging it now

**Pull request merged**